### PR TITLE
Prototype means schema

### PR DESCRIPTION
**Pull request by timpeat**
## Description of change
Prototype schema for Means attributes

## Link to relevant ticket
[CRIMAP-658](https://dsdmoj.atlassian.net/browse/CRIMAP-658)

## Additional notes

Also introduces a shared example to ensure that a struct's schema is aligned with that in json_schema proper.

[CRIMAP-658]: https://dsdmoj.atlassian.net/browse/CRIMAP-658?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: lib/laa_crime_schemas/types/types.rb [hunk 1]
```diff
@@ -118,6 +156,19 @@ module Types
...
+    SavingsType = String.enum(*%w[
+                                cash
+                                other_person_account
+                                certificates
+                                shares
+                                trusts
+                                policies
+                              ])
+
+    EmploymentStatusType = String.enum(*%w[employed self-employed none])
```
Comment (arthurashman) [inline lib/laa_crime_schemas/types/types.rb:170]: Worth adding in the others we're collecting here?
partnership
director
private_shareholder

File: spec/fixtures/application/1.0/means.json [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      {
+				"type": "child",
+				"amount": 3990,
+				"frequency": "month"
+			}
+		],
+		"employment_details": {
+			"paye": [
+				{
+				  "amount": 1320000,
```
Comment (arthurashman) [inline spec/fixtures/application/1.0/means.json:15]: You ended up going for pence here right?? Just wondering if there's anywhere sensible to record that decision. Not saying here is the place. Just the salary made me think about it

File: lib/laa_crime_schemas/types/types.rb (hunk 1 above)
Comment (timpeat) [reply lib/laa_crime_schemas/types/types.rb:170]: Perhaps in addition to the employment_status we have  an array "employment_types" to record all the employment types of the client. E.g. ``` "employment_types": ["director", "private_shareholder", "paye"]

File: lib/laa_crime_schemas/structs/means_details.rb [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+      attribute? :outgoings_details do
+        attribute :outgoings, Types::Array.of(Base) do
+          attribute :type, Types::OutgoingsType
+          attributes_from Amount
+        end
+      end
+
+      attribute? :dependents, Types::Array.of(Base) do
+        attribute :date_of_birth, Types::JSON::Date
```
[the remark this answers is not in the document]
Comment (timpeat) [reply lib/laa_crime_schemas/structs/means_details.rb:84]: Good point, I'll revert this back.

File: lib/laa_crime_schemas/structs/means_details.rb [hunk 4]
```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+
+module LaaCrimeSchemas
+  module Structs
+    class MeansDetails < Base
+      attribute :employment_status, Types::EmploymentStatusType
+
+      attribute :income_details do # rubocop:disable Metrics/BlockLength
+        attribute :benefits, Types::Array.of(Base) do
+          attribute :type, Types::OtherBenefitType
```
Comment (willmcb) [inline lib/laa_crime_schemas/structs/means_details.rb:10]: Whats the reason for 'Other' in the type name here? Is this based on the way it is named in the UI

File: lib/laa_crime_schemas/types/types.rb (hunk 1 above)
Comment (timpeat) [reply lib/laa_crime_schemas/types/types.rb:170]: @arthurashman  if it is okay, I'd rather leave this (how we capture employment income sources) until the self-employed journey is finessed.

File: lib/laa_crime_schemas/structs/means_details.rb (hunk 4 above)
Comment (timpeat) [reply lib/laa_crime_schemas/structs/means_details.rb:10]: These are a subset of benefits that are combined to produce the single value "other_benefits_amount" used by MAAT.

File: lib/laa_crime_schemas/types/types.rb (hunk 1 above)
Comment (arthurashman) [reply lib/laa_crime_schemas/types/types.rb:172]: Ok, all good. We don't need it just yet anyhow

**Review by arthurashman — APPROVED** [review]
👍 Great work - this is super complex, thanks for unpicking!

**Pull request merged**